### ISSUE TITLE
base round-trip errors

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#4723]
Syntax:
- [ ] #5310
- [ ] #4725
- [ ] #4726 (repro'ed aug 29 2024)
  - [ ] Unit.doc (#4726)
- [x] #5317
- [x] #4727
- [ ] docs.basicDataTypes (duplicate of #4725)
- [x] IO.net.Socket.doc
- [x] #4728
  - [ ] metadata.licenseTypes.bsd3.doc (#4728)
  - [ ] metadata.licenseTypes.mit.doc (#4728)
- [x] Optional.getOrElse.doc
- [x] #4729
- [x] test.laws.distributive.doc
- [x] #4731
- [x] Void.doc
- [ ] #4733
- [ ] #5306

Language change?
- [x] abilities.forever (blocked by #4725)
- [x] data.List.insertSortedDistinct
- [x] data.List.split
- [ ] #5315
- [x] data.Map.internal.putWithR
- [x] data.NatSet.Nonempty.intersect
- [x] data.Weighted.filter
- [x] data.Weighted.flatMap
- [x] data.Weighted.map
- [ ] data.Weighted.weight (blocked by hash mismatch on Share)
- [x] GUID.toBase16
- [x] IO.concurrent.STM.TMap.delete
- [x] IO.net.Connection.closer.modify
- [x] IO.net.Connection.closer.set
- [x] IO.net.Connection.receiver.modify
- [x] #4724
- [x] IO.net.Connection.send.modify
- [x] IO.net.Connection.send.set
- [x] math.Natural.+
- [x] Search.exact
- [x] Search.lub
- [x] test.internals.v1.Test.modifyScope
- [x] test.internals.v1.Test.modifyStatus
- [x] Text.findFirst

Unsure:
- [x] abilities.repeat
- [x] abilities.Store.local!.examples.ex1
- [x] data.List.foreach
- [x] data.List.foreach_
- [ ] #4736
- [ ] data.List.Nonempty.somes.tests.ex1 references an unnamed hash, but unnamed hashes aren't accepted by the parser. (hashes are accepted if they have some name in the branch, rejected if they don't, even if they match a unique prefix.)
- [ ] #5307

Syntax Highlighting:
- [x] abilities.Random.intIn.tests.fairness
- [x] abilities.Random.natIn.tests.fairness
- [x] Float.doc
- [x] Float.fromText.doc
- [x] Int.diff.doc
- [x] Int.doc
- [x] Int.fromText.doc
- [x] IO.net.HostName.doc
- [x] IO.net.Socket.doc
- [x] math.Natural.docs.arithmetic
- [x] Nat.doc
- [x] Nat.fromText.doc
- [x] time.Duration.deprecated.displayDuration.tests
- [x] time.patterns.rfc2822Offset.doc
- [x] time.UTCOffset.fromBasicISO8601.doc
- [x] time.UTCOffset.fromIso8601.doc
- [x] time.UTCOffset.fromRFC2822.doc
